model/event/subscriptions.go

```diff
@@ -214,9 +214,18 @@ func FindSubscriptionByID(id string) (*Subscription, error) {
 		subscriptionIDKey: id,
 	}), &out)
 	if err == mgo.ErrNotFound {
-		return nil, nil
-
-	} else if err != nil {
+		if bson.IsObjectIdHex(id) {
+			err = db.FindOneQ(SubscriptionsCollection, db.Query(bson.M{
+				subscriptionIDKey: bson.ObjectIdHex(id),
```

Comment:
I think it would be better to use $or for this in the query above, or migrate everything to strings.

Reply:
Good point, fixed